kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_support/compile_utils/component.py

```diff
@@ -58,6 +58,9 @@ def set_s3_env_vars_to_component(
         """
 
         if prefix is not None:
-            for env_name, _ in env2key.items():
-                env2key[prefix + "_" + env_name] = env2key.pop(env_name)
-        kubernetes.use_secret_as_env(task=task, secret_name="s3-secret", secret_key_to_env=env2key)
+            for secret_key, _ in env2key.items():
```

Comment:
why do you need this replacement?
if the prefix is None, you will get strange names, started form "_".